### PR TITLE
LPD-22671 - Reverts code from LPS-128286

### PR DESCRIPTION
**Pull request by thallescezar**
Related issue: https://liferay.atlassian.net/browse/LPD-22671

I'm reverting this [commit](https://github.com/liferay/liferay-portal-ee/commit/81d79e1bf6e27e799982b0f960ed49d441a677c2). 
I believe that the team that made this change didn't realize that this functionality still in use for DDL and that was causing the reported problem.

### CODE REVIEW

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
ci:test:sf

**Review by pedro-oliveira446 — APPROVED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js [hunk 1]
```diff
@@ -122,7 +117,7 @@ AUI.add(
 			'</label>' +
 			'</div>';
 
-		const TPL_SEPARATOR = '<hr class="separator" />';
+		const TPL_SEPARATOR = '<div class="separator"></div>';
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js:120]: @thallescezar why did you make this change?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js [hunk 2]
```diff
@@ -222,7 +217,7 @@ AUI.add(
 
 					const input = instance.get('boundingBox').one('input');
 
-					if (/#[A-F\d]{6}/.test(input.val().toUpperCase())) {
+					if (/#[A-F\d]{6}/.test(input.val())) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js:220]: @thallescezar why did you make this change?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js [hunk 3]
```diff
@@ -254,9 +249,9 @@ AUI.add(
 					const input = instance.get('boundingBox').one('input');
 
 					if (input) {
-						const val = input.val().toUpperCase();
+						const val = input.val();
 
-						if (/#[A-F\d]{6}/.test(val) || val === '') {
+						if (/#[A-F\d]{6}/.test(val)) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js:254]: @thallescezar why did you make this change?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js [hunk 4]
```diff
@@ -598,6 +788,7 @@ AUI.add(
 		const customCellEditors = [
 			ColorCellEditor,
 			DLFileEntryCellEditor,
+			JournalArticleCellEditor,
 			IntegerCellEditor,
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/custom_fields.js:792]: @thallescezar sort alphabetically, i.e., `JournalArticleCellEditor` after `IntegerCellEditor`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js [hunk 5]
```diff
@@ -1019,6 +1019,12 @@ AUI.add(
 				repeat() {
 					const instance = this;
 
+					const field = instance.getFieldDefinition();
+
+					if (field.type === 'select') {
+						field.options.shift();
+					}
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js:1027]: @thallescezar why did you make this change?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js [hunk 6]
```diff
@@ -1661,7 +1667,9 @@ AUI.add(
 						'documentLibrarySelectorURL'
 					);
 
-					let retVal = instance.getDocumentLibraryURL('file');
+					let retVal = instance.getDocumentLibraryURL(
+						'com.liferay.item.selector.criteria.file.criterion.FileItemSelectorCriterion'
+					);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js:1672]: @thallescezar why did you make this change?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js [hunk 7]
```diff
@@ -1834,13 +1842,327 @@ AUI.add(
...
+							label: Liferay.Language.get('all'),
+							layoutId: 0,
+							privateLayout,
+						};
+
+						return [layoutsRoot];
+					},
+				},
+			},
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js:2157]: @thallescezar why did you add this new code?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js [hunk 8]
```diff
@@ -1834,13 +1842,327 @@ AUI.add(
...
 			EXTENDS: Field,
 
 			prototype: {
 				_handleClearButtonClick() {
 					const instance = this;
 
+					instance._clearedModal = true;
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/legacy/ddm_form.js:2164]: @thallescezar why did you add this new code?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
@thallescezar, I noticed some changes that were not in the reverted [commit](https://github.com/liferay/liferay-portal-ee/commit/81d79e1bf6e27e799982b0f960ed49d441a677c2).

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by thallescezar** [comment]
@carolmariaabb I redid the cherry-pick and it solves all of your questions.

**Comment by thallescezar** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Pull request closed without merging**